platform/default/include/mbgl/storage/offline_database.hpp

```diff
@@ -96,6 +94,9 @@ class OfflineDatabase : private util::noncopyable {
     std::exception_ptr pack();
     void runPackDatabaseAutomatically(bool autopack_) { autopack = autopack_; }
 
+    // For testing only
+    void reopenDatabaseReadOnly();
```

Comment:
`reopenDatabaseForTests()` ?

Reply:
Make it a constructor parameter or a dedicated constructor?

Reply:
`reopenDatabaseReadOnlyForTesting` soungs great. Making a separated constructor for it will make us carry this flag a long way...